javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -12,15 +12,39 @@ module ExceptionXss {
   import Xss as Xss
   private import semmle.javascript.dataflow.InferredTypes
 
+  /**
+   * Gets the name of a method that does not leak taint from its arguments if an exception is thrown by the method.
+   */
+  private string getAnUnlikelyToThrowMethodName() {
+    result = "getElementById" or
+    result = "indexOf" or
+    result = "stringify" or
+    result = "assign" or
+    // fs methods. (The callback argument to the async functions are vulnerable, but its unlikely that the callback is the user-controlled part).
+    result = "existsSync" or
```

Comment:
For these fs methods, you may be able to make use of the externs like we do here: https://github.com/Semmle/ql/blob/c4d21633215700dae90f148ec1ef1182787b59b3/javascript/ql/src/semmle/javascript/frameworks/NodeJSLib.qll#L400-L418